File: node-client-migration/src/es-client-main.js
// https://www.npmjs.com/package/@elastic/elasticsearch/v/7.13.0
// Compatibility https://opensearch.org/docs/latest/clients/index/

require('dotenv').config();
const { Client } = require('@elastic/elasticsearch');

async function run(client, index = 'game-of-thrones') {
  // Let's start by indexing some data
  await client.index({
    index: index,
    body: {
      character: 'Ned Stark',
      quote: 'Winter is coming.',
    },
  });

  await client.index({
    index: index,
    body: {
      character: 'Daenerys Targaryen',
      quote: 'I am the blood of the dragon.',
    },
  });

  await client.index({
    index: index,
    body: {
      character: 'Tyrion Lannister',
      quote: 'A mind needs books like a sword needs a whetstone.',
    },
  });

  // here we are forcing an index refresh, otherwise we will not
  // get any result in the consequent search
  await client.indices.refresh({ index: index });

  // Let's search!
  const { body } = await client.search({
    index: index,
    body: {
      query: {
        match: { quote: 'winter' },
      },
    },
  });

  const count = body.hits.hits.length;
  return {
    success: count > 0,
    count: count,
    message: `Found ${count} record(s)`,
  };
}

const index = `game-of-thrones-${Math.trunc(Math.random() * 1000)}`;

// Test ES client against ES cluster
const ES_SERVICE_URI = process.env.ES_SERVICE_URI;
const esClient = new Client({ node: ES_SERVICE_URI });
run(esClient, index).then(console.log).catch(console.log);

// Test ES client against OS cluster
const OS_SERVICE_URI = process.env.OS_SERVICE_URI;
const osClient = new Client({ node: OS_SERVICE_URI });
run(osClient, index).then(console.log).catch(console.log);

File: python-client-migration/python_client_migration/es_client_main.py
# https://elasticsearch-py.readthedocs.io/en/v7.16.3/
# Compatibility https://opensearch.org/docs/latest/clients/index/

import os
from random import randint
from dotenv import load_dotenv
from elasticsearch import Elasticsearch

load_dotenv()


def run(client, index="game-of-thrones"):
    ## Let's start by indexing some data
    client.index(
        index=index,
        body={
            "character": "Ned Stark",
            "quote": "Winter is coming.",
        },
    )

    client.index(
        index=index,
        body={
            "character": "Daenerys Targaryen",
            "quote": "I am the blood of the dragon.",
        },
    )

    client.index(
        index=index,
        body={
            "character": "Tyrion Lannister",
            "quote": "A mind needs books like a sword needs a whetstone.",
        },
    )

    response = client.search(
        body={
            "query": {
                "match": {"quote": "winter"},
            },
        },
        index=index,
    )

    count = len(response["hits"]["hits"])
    return {"success": count > 0, "count": count, "message": f"Found {count} record(s)"}


if __name__ == "__main__":
    # Generate random index
    index = "game-of-thrones-" + str(randint(0, 500))

    # Test ES client against ES cluster
    ES_SERVICE_URI = os.getenv("ES_SERVICE_URI")
    client_against_elasticsearch = Elasticsearch(ES_SERVICE_URI, use_ssl=True)
    print(run(client_against_elasticsearch))

    # Test ES client against OS cluster
    OS_SERVICE_URI = os.getenv("OS_SERVICE_URI")
    client_against_opensearch = Elasticsearch(OS_SERVICE_URI, use_ssl=True)
    print(run(client_against_opensearch))
